pkg/addon-operator/operator.go

```diff
@@ -765,15 +766,19 @@ func (op *AddonOperator) TaskHandler(t sh_task.Task) queue.TaskResult {
 					BindingType:              OnKubernetesEvent,
 					BindingContext:           info.BindingContext,
 					AllowFailure:             info.AllowFailure,
-					ReloadAllOnValuesChanges: false, // Ignore global values changes
-					KubernetesBindingId:      kubernetesBindingId,
-					WaitForSynchronization:   info.WaitForSynchronization,
+					ReloadAllOnValuesChanges: false, // Ignore global values changes in global Synchronization tasks.
+					KubernetesBindingId:      kubernetesBindingID,
+					WaitForSynchronization:   info.KubernetesBinding.WaitForSynchronization,
+					MonitorIDs:               []string{info.KubernetesBinding.Monitor.Metadata.MonitorId},
+					ExecuteOnSynchronization: info.KubernetesBinding.ExecuteHookOnSynchronization,
 				})
+
+			// Ignore "waitForSynchronization: true" if Synchronization is not required.
 			if info.QueueName == t.GetQueueName() {
 				mainSyncTasks = append(mainSyncTasks, newTask)
 			} else {
-				if info.WaitForSynchronization {
-					waitSyncTasks[kubernetesBindingId] = newTask
+				if info.KubernetesBinding.WaitForSynchronization && info.KubernetesBinding.ExecuteHookOnSynchronization {
```

Comment:
Oh, I thought that `waitForSynchronization` controls the moment when we start firing monitor events. It can be tricky that `executeHookOnSynchronization` also affects it.

Reply:
`waitForSynchronization: true` is to postpone `beforeHelm` and `beforeAll` until "Synchronization" is done in parallel queues. It is not an obvious name, yes ;)